Source/WebCore/platform/graphics/haiku/GraphicsContextHaiku.cpp

```diff
@@ -396,19 +411,31 @@ void GraphicsContextHaiku::drawFocusRing(const Vector<FloatRect>& rects, float w
     m_view->PopState();
 }
 
-void GraphicsContextHaiku::drawLinesForText(const FloatPoint& point, float, const DashArray& widths, bool printing, bool doubleUnderlines, WebCore::StrokeStyle)
+void GraphicsContextHaiku::drawLinesForText(const FloatPoint& point,
+    float thickness, const DashArray& widths, bool printing,
+    bool doubleUnderlines, WebCore::StrokeStyle style)
 {
-    if (widths.size() <= 0)
+    if (widths.isEmpty())
         return;
 
+    Color lineColor(strokeColor());
+    FloatRect bounds = computeLineBoundsAndAntialiasingModeForText(
+        FloatRect(point, FloatSize(widths.last(), thickness)),
+        printing, lineColor);
+    float y = bounds.center().y();
+
+    float oldSize = m_view->PenSize();
+    m_view->SetPenSize(bounds.height());
+
     // TODO would be faster to use BeginLineArray/EndLineArray here
-    // TODO in Cairo, these are not lines, but filled rectangle? Whats the thickness?
     for (size_t i = 0; i < widths.size(); i += 2)
     {
         drawLine(
```

Comment:
why not call StrokeLine directly here? It would simplify the code in drawLine to not need the special case?